openvpn/process.go

```diff
@@ -24,24 +24,20 @@ import (
 
 	"github.com/mysteriumnetwork/node/openvpn/config"
 	"github.com/mysteriumnetwork/node/openvpn/management"
+	"github.com/mysteriumnetwork/node/openvpn/tunnel"
 )
 
-type tunnelSetup interface {
```

Comment:
You reverted 2 improvements
- Interface should be on caller side. Not in implementation
- Implementation factory should return not interface but exact structs `new() return *SetupTunInterface`

Reply:
@tadovas, @sofokkl we kind of agreed to continue like this. But now we have 2 conventions and that missleads

Reply:
In this case the only option is to return interface, if we want to choose os specific implementation on compile time, and hide those details from process (generic tun setup on mac, win) (specific tun on linux), 
On the other hand, we can have two functions with the same name and conditionaliy compiled, but with different return values, i.e. NewTunnelSetup() * LinuxTunnel NewTunnelSetup() * GenericTunnel, but I think it will hurt readability as the same function will return different result on different OS. I would vote for interface declaration on implementor side in this particular case.

Reply:
Return structures depend on interfaces is generic golang approach and it's fine, but there are edge cases where you have to choose :) I think it's the case

Reply:
> ..if we want to choose os specific implementation on compile time.. 

To solve this i moved conditional factory to upper layer:
+ caller defines interface `TunnelSetup` https://github.com/mysteriumnetwork/node/commit/84b2b3b3d698ebae264f8bb041c6478faea7054a

+ caller decides implementation to go for Linux/ARM because he knows where he is compiling

+ Caller can construct `SetupTunInterface` in ARM. And now is made private

This was idea - to make `tunnel` package independent from our process so that someone can use just Tunnel seperately.